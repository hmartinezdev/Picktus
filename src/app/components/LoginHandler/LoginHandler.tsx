import Button from '@components/Button';
import Input from '@components/Input';
import SubscribeHandler from '@components/SubscribeHandler';
import colors from '@constants/colors';
import React, { Component } from 'react';

class LoginHandler extends Component {
  public render(): React.ReactElement<LoginHandler> {
    return (
      <div className="container">
        <SubscribeHandler />
        <div className="login">
          <Input placeholder="Mail" />
          <Input placeholder="Password" type="password" />
          <Button text="Login" />
        </div>
        <style jsx>{`
          .container {
            position: relative;
          }

          .login {
            display: flex;
            flex-direction: column;
            align-items: center;
            background-color: ${colors.primary};
            min-width: 22rem;
            padding: 1.5rem;
            box-sizing: border-box;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }
        `}</style>
      </div>
    );
  }
}
export default LoginHandler;
