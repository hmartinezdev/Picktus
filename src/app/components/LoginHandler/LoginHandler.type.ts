export interface IStringMap {
  [s: string]: string;
}
export interface ILoginHandlerState {
  inputs: IStringMap;
}
