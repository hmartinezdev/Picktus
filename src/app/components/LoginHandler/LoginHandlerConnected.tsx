import { classicLogin, facebookLogin, googleLogin, twitterLogin } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import LoginHandler from './LoginHandler';

export default connect(
  undefined,
  { classicLogin, googleLogin, facebookLogin, twitterLogin }
)(LoginHandler);
