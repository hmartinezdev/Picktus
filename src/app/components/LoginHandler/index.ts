import LoginHandler from './LoginHandler';

export default LoginHandler;
