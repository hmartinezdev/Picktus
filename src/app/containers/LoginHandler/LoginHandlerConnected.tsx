import { signin } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import LoginHandler from './LoginHandler';

export default connect(
  undefined,
  { signin }
)(LoginHandler);
