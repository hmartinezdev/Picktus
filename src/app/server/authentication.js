const admin = require('firebase-admin');

async function authenticate(req, res) {
  const sessionCookie = req.cookies.session || '';
  // Verify the session cookie. In this case an additional check is added to detect
  // if the user's Firebase session was revoked, user deleted/disabled, etc.

  try {
    const result = await admin.auth().verifySessionCookie(sessionCookie, true /** checkRevoked */);
    return result;
  } catch (e) {
    return undefined;
  }
}

module.exports = authenticate;
