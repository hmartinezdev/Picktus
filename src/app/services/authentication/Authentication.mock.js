export const validUserCredentials = {
  user: {
    getIdToken: () => {
      return new Promise((resolve) => resolve('idToken'));
    },
  },
};

export const incorrectUserCredentials = {};
