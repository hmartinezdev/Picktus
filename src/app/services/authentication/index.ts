import Authentication from './Authentication';
export * from './Authentication.type';

export default Authentication;
