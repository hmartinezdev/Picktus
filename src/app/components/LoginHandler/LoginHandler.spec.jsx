import enzyme, { shallow, mount } from 'enzyme';
import React from 'react';
import LoginHandler from './LoginHandler';
import { SigninMethods } from '@services/authentication';

let wrapper;
let mountWrapper;

const setup = (props = {}) => shallow(<LoginHandler {...props} />);
const mountSetup = (props = {}) => mount(<LoginHandler {...props} />);

describe('<LoginHandler />', () => {
  beforeEach(() => {
    wrapper = setup();
  });

  test('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('onChange', () => {
    test('it should change the input value in the component state', () => {
      const instance = wrapper.instance();
      const spy = jest.spyOn(instance, 'setState');
      instance.onChange({ target: { name: 'test', value: 'test' } });
      expect(spy).toHaveBeenCalled();
      expect(spy).toHaveBeenCalledWith({ inputs: { test: 'test' } });
    });

    test('it should not change the input value if the name is open', () => {
      const instance = wrapper.instance();
      const spy = jest.spyOn(instance, 'setState');
      instance.onChange({ target: { name: 'open', value: 'test' } });
      expect(spy).toHaveBeenCalledTimes(0);
    });
  });

  describe('login', () => {
    const spyLogin = jest.fn();
    let instance;

    beforeEach(() => {
      wrapper = setup({ signin: spyLogin });
      instance = wrapper.instance();
    });

    afterEach(() => {
      jest.resetAllMocks();
    });

    test('onLoginClick should call the signin method in props with methods.CLASSIC and password and email as options', () => {
      instance.setState({ inputs: { password: 'test', email: 'mail' } });
      instance.onLoginClick();
      expect(spyLogin).toHaveBeenCalledWith(SigninMethods.CLASSIC, { password: 'test', email: 'mail' });
    });

    test('onGoogleClick should call the signin method in props with methods.GOOGLE ', () => {
      instance.onGoogleClick();
      expect(spyLogin).toHaveBeenCalledWith(SigninMethods.GOOGLE, {});
    });

    test('onFacebookClick should call the signin method in props with methods.FACEBOOK ', () => {
      instance.onFacebookClick();
      expect(spyLogin).toHaveBeenCalledWith(SigninMethods.FACEBOOK, {});
    });

    test('onTwitterClick should call the signin method in props with methods.TWITTER ', () => {
      instance.onTwitterClick();
      expect(spyLogin).toHaveBeenCalledWith(SigninMethods.TWITTER, {});
    });
  });
});
