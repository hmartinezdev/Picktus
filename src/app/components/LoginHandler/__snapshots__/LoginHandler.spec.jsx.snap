// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginHandler /> should render properly 1`] = `
<div
  className="container"
>
  <div
    className="login"
  >
    <Input
      name="email"
      onChange={[Function]}
      placeholder="Mail"
      type="text"
    />
    <Input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <Button
      onClick={[Function]}
      text="Login"
    />
    <div
      className="socialButtons"
    >
      <Button
        onClick={[Function]}
        text=""
      >
        <Component
          className="social"
        />
      </Button>
      <Button
        onClick={[Function]}
        text=""
      >
        <Component
          className="social"
        />
      </Button>
      <Button
        onClick={[Function]}
        text=""
      >
        <Component
          className="social"
        />
      </Button>
    </div>
  </div>
  <Connect(SubscribeHandler) />
  <style
    jsx={true}
  >
    
          .container {
            position: relative;
          }

          .login {
            display: flex;
            flex-direction: column;
            align-items: center;
            background-color: #2d3047;
            min-width: 22rem;
            padding: 1.5rem 0;
            box-sizing: border-box;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .socialButtons {
            display: flex;
            justify-content: space-between;
            width: 75%;
          }

          :global(.socialButtons.socialButtons &gt; button:nth-child(2)) {
            margin: 0.7rem 1.4rem;
          }
        
  </style>
</div>
`;
